### ISSUE TITLE
DSD/DSF support

### ISSUE THREAD

**Issue by sat3ll** [clementine-player/Clementine#6407]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [ x] I checked the issue tracker for similar issues
- [ x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [ ] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Fedora Linux
- Clementine version: 1.3.1 (flatpak from Flathub)

### Expected behaviour / actual behaviour
Playing `.dsf` / DSD audio files.

**Comment by v-fox** [comment]
Personally, I'm not interested in playing DSD files even though my DAC is capable of doing so but I just want to say that it would be _reeealy_ unorthodox for a player to play them correctly. Because there is an easy way and the right way. The easy way is converting them into PCM stream and feeding it as any other audio BUT that would defeat the whole point. The right way is to feed DSD stream directly into DAC. BUT
* you need DSD-capable DAC with 'Format: SPECIAL DSD_U32_BE' in /proc/asound/\<DAC>/stream0
* you need the kernel driver that advertises it, like snd-usb-audio
* you need manually configure ALSA in ~/.asoundrc to allow using it without conversion, like:
```
# https://github.com/lintweaker/xmos-native-dsd
audio_output {
  type "alsa"
  name "USB XMOS DAC"
  device "hw:x20"
  # Enable native DSD playback
  dsd_native "yes"
  # Disable DSD-over-PCM conversion
  dsd_usb "no"
  # Select 32-bit DSD_U32_BE output format
  dsd_native_type "2"
}
```
* and then you'll need to deal with that interfering with PA or JACK, if you use them

**Comment by sat3ll** [comment]
"Easy-way" might be preferable to "no-way" since in cases where the user library contains DSD files, converting them specifically to get them recognized by Clementine would needlessly clutter and occupy more storage.

**Comment by v-fox** [comment]
> "Easy-way" might be preferable to "no-way" since in cases where the user library contains DSD files, converting them specifically to get them recognized by Clementine would needlessly clutter and occupy more storage.

@sat3ll Yes. But "easy-way" is just realtime PCM conversion into RAM. So, instead of keeping converted copy on permanent storage medium, you just will be deleting it after every playback.
